### DISCUSSION TITLE
Mapping with Dataview queries

### DISCUSSION THREAD

**Discussion by charle36** [chrisgurney/obsidian-note-toolbar#64] [Ideas]
Awesome plugin! Thanks for this great contribution to Obsidian!

If I may offer a suggestion, I think an interesting addition to the current features would be to make mapping through Dataview queries possible. This would allow for mapping throught tags and other properties (I'm not fond of tags personally), thus allowing for different workflows.

Tell me what you think of the idea!

**Comment by chrisgurney** [comment]
@charle36 Thanks for the suggestion! I did a little bit of exploratory work with supporting Dataview when I started the plugin, but didn't make it too far. Note though this one _may_ have to wait until the Obsidian team has implemented their plans for Dataview-like functionality (_Dynamic views_ on [the roadmap](https://obsidian.md/roadmap/)).

What use cases did you have in mind with Dataview?

**Reply by charle36** [reply]
With Dataview queries (or, at least, tag queries) you can support horizontal structures.

For instance, if I have a type of notes that is identified by a property and I want it to have it's own toolbar, I can't do it with the current mapping options. 

In my specific use case, I would like to have a different toolbar for project notes, but they don't have a folder just for them in my system. I could switch, but a lot of people prefer horizontal organisation and Obsidian is a good tool for that kind of structure.

> Note though this one _may_ have to wait until the Obsidian team has implemented their plans for Dataview-like functionality (_Dynamic views_ on [the roadmap](https://obsidian.md/roadmap/)).

Yeah, its probably better to wait. We are all waiting on that update! xD

**Reply by chrisgurney** [reply]
Hmm, so what about the ability to map tags to toolbars? That seems interesting to me, as it's a solution that's _not_ specific to Dataview.

Perhaps the folder mappings section becomes more generic, and the left column becomes a choice between a folder and a property rule (perhaps starting with just looking at the `tags` prop to start).

**Reply by charle36** [reply]
These are all good ideas. I just thought about the integration with Dataview first since is seemed like the easiest way to offer the most options, but I'm not familiar with how it would work.

I think tags are a good start for sure!

**Reply by chrisgurney** [reply]
@charle36 Mapping tags to toolbars is currently number 3 or 4 on my list, so it's in the queue.

Just to confirm, would this feature solve your particular need?

**Reply by charle36** [reply]
Yes, that would work. Thanks!

**Reply by FeralFlora** [reply]
> For instance, if I have a type of notes that is identified by a property and I want it to have it's own toolbar, I can't do it with the current mapping options.
> In my specific use case, I would like to have a different toolbar for project notes

@charle36 Isn't this already possible, if you use the property option to define which toolbar to associate with a given note? 
https://github.com/chrisgurney/obsidian-note-toolbar/wiki/Defining-where-to-show-toolbars#property

You don't have to use folder mappings. You can add this to your frontmatter instead:

```
---
notetoolbar: Project toolbar
---
```

**Reply by chrisgurney** [reply]
@FeralFlora This is true, the `notetoolbar` prop is a good workaround for now, but requires this prop to be updated each time the note needs a new toolbar, and adds an additional line of YAML.

I think the idea of tags mapping to toolbars is a valuable one, for the reason(s) originally mentioned. It's on my list for 1.9, once I'm happy with the 1.8 release (settings UI improvements) and get it out the door... _hopefully_ in the next few days.

**Reply by FeralFlora** [reply]
Right, I agree that this idea is certainly valuable in its own right. I was just pointing out that the reasoning that brought the suggestion about seems to be rooted in a misunderstanding about the current possibility of using properties to specify toolbars.

> support horizontal structures.

This is already a reality, is all I'm saying.

**Reply by chrisgurney** [reply]
@FeralFlora Appreciate it! I love when the community solves each other's problems. 😄

**Reply by charle36** [reply]
I played with the propriety option and I realized that I did overlook this feature. 

It's actually more powerful than you give it credit for. You can set it to "tags", then rename your toolbars to the name of your tag to get tag based mapping. It doesn't seems to lead to any problem and it makes my initial request almost obsolete (it only works with tags in YAML; it requires a correspondance between the toolbars' name and tags).

Thanks to both of you!

**Comment by chrisgurney** [comment]
See also [this discussion](https://github.com/chrisgurney/obsidian-note-toolbar/discussions/110) about scripting and an API.

**Comment by chrisgurney** [comment]
@charle36 The ability to map toolbars based on tags and properties (and a lot more) is now [generally available](https://github.com/chrisgurney/obsidian-note-toolbar/releases/tag/1.35.10).

_If you give it a try, please let me know what you think!_
